### PR TITLE
refactor code with hooks

### PR DESCRIPTION
**Pull request by Kwentsir**
In this milestone, the code of the Calculator component (components/Calculator.js) was refactored from a class-based component to a functional component with hooks.

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,108 +1,100 @@
...
+  const handleCalculate = (label) => {
+    const { total, next, operation } = calcState;
     const newObj = calculate({ total, next, operation }, label);
-    this.setState({
+    setCalcState({
       total: newObj.total,
       operation: newObj.operation,
       next: newObj.next,
     });
   };
```
Comment (rindrajosia) [inline src/components/calculator.js]: - [ ] [REQUIRED] Please make sure that the next operand is displayed on the screen when users enter it.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:

## Required Changes 💪 

- I am able to make operation

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.
- Please don't disable linters rules as linter is there in order to help you to have a good code base quality.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js [hunk 2]
```diff
@@ -1,108 +1,105 @@
...
+  const handleCalculate = (label) => {
+    const { total, next, operation } = prevState;
     const newObj = calculate({ total, next, operation }, label);
-    this.setState({
+    setState({
       total: newObj.total,
       operation: newObj.operation,
       next: newObj.next,
     });
   };
```
Comment (mahbubul14) [inline src/components/calculator.js]: - [ ] The basic calculator operators (**ex: '+', '-', '/')** of your calculator app are not displayed on the display bar when a user enters it. As the previous reviewer suggested, kindly work on this.

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @Kwentsir ,

Good job so far! :+1: 
You still need to work on some issues to go to the next project, but you are almost there!

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review :arrow_down: 

###  To Highlights
:heavy_check_mark: No linters error.
:heavy_check_mark: Descriptive Readme.

### Optional suggestions

- [ ] The commit message should be written in the imperative tense and the first also should be capitalized. `Ex: "Fix bug" instead of "bug fix".` [For more details.](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by bimbolabuari — APPROVED** [review]
Hi @Kwentsir,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**